### ISSUE TITLE
Turkish present tense covert copula overrides the features of its host.

### ISSUE THREAD

**Issue by TalhaBedir** [UniversalDependencies/docs#774]
Traditionally Turkish present tense copular paradigm is as follows (covert copula is depicted as [cop]):

1. Ben öğrenci-[cop]-yim "I am a student"
2. Sen öğrenci-[cop]-sin "You are a student"
3. O öğrenci-[cop] "He is a student" 
4. Biz öğrenci-[cop]-yiz "We are students"
5. Siz öğrenci-[cop]-siniz "You are students"
6. Onlar öğrenci-[cop]-(ler) "They are students"

So in sentence (4), the features of the predicate _öğrenciyiz_ with the current annotation style would be like: Case=Nom|Number=Plur|Person=1

This means that the numeral and personal features of _öğrenci_ which are Number=Sing|Person=3 are gone and the copular part's features Number=Plur|Person=1 are inserted. In a way, nominal features are _overridden_.

I propose Number[cop] and Person [cop] features that are created in a similar vein with Number[psor] and Person[psor]. The latter ones were for possessor morphology inflected onto nouns, and the former ones, i.e. the ones I propose are for copular morphology on nominal and adjectival predicates.

If we are to accept this, then features of the predicate in sentence (4) would be like the following:

Case=Nom|Number=Sing|Number[cop]=Plur|Person=3|Person[cop]=1

**Comment by coltekin** [comment]
Current Turkish/Turkic treebanks split the copular suffixes (attached to nouns). So, this conflict is avoided.

This is not necessarily the best approach, and it has been discussed extensively before (for example, #125, #377, #706, but more outside the UD issues) without any clear verdict. Nevertheless, I suggest following the existing treebanks, even if that does not feel completely right. Not splitting the copular suffixes from nouns would introduce another inconsistency among Turkish treebanks (which suffer from many intentional/unintentional inconsistencies already).

If you'd like to follow the no-split path, I think we should discuss it further here (again), and if it seems to be the best option, we should consider unifying annotations in other treebanks as well.

**Comment by ftyers** [comment]
And other Turkic too...

**Comment by TalhaBedir** [comment]
We currently split two morphology in our treebank: cliticized copulas (-(y)DI, -(y)mIş, -(y)ken, -(y)SA, including the ones after TAM-1 markers, e.g. _gitmişti_ (he had gone)) and two types of "ki"s (the one that makes nouns like _bizimki_ (ours), and adjectives like _evdeki_ NP (the NP that is at home))

We proposed that split path since harmony is obviously desired, though, we are not sure if the present tense copula has parallel morphology with the others we split. Because the other copulas are actually splittable: _arabaydı_ (it was a car) and _araba idi_ are both acceptable and have the same meaning. However, the present tense copula (a) inseparable, (b) never surfaces in any form. Because the paradigm -yim, -sin etc. above is not actually demonstrating any copula, they are merely phi-feats attached to an invisible syntactic unit that is argued to be copula. Therefore we decided to include split way for -(y)DI, (y)mIş etc. and non-split way for the present tense copula.

I think there needs to be at least a hint in the system that distinguishes between splittable and non-splittable copulas, but hey that's just me. A relevant and broader question would be how important the harmony is in and among treebanks and how important it is to annotate complying with morphosyntax of Turkish.

**Comment by dan-zeman** [comment]
> A relevant and broader question would be how important the harmony is in and among treebanks and how important it is to annotate complying with morphosyntax of Turkish.

Both are important and they should not be put in conflict with each other. If all previously existing Turkish treebanks take approach A and you believe that it complies less with Turkish grammar than approach B, then ideally you should first persuade the providers of the existing Turkish treebanks to change their approach, rather than simply publishing another treebank that annotates the same thing differently. Inconsistencies are pure evil. They prevent comparing data, or applying tools trained on one dataset to another dataset. (And this also extends to other related languages, if it is clear that the phenomenon works exactly the same way there.)

**Comment by jonorthwash** [comment]
> they are merely phi-feats attached to an invisible syntactic unit that is argued to be copula

Could you explain what you mean by this, and what relevance it has to UD annotation?

I would argue that a split is more consistent within Turkish and across Turkic languages.  You seem to argue the former by saying the following:

> we decided to include split way for -(y)DI, (y)mIş etc. and non-split way for the present tense copula.

This seems to be inconsistent.  Consider also that the future form of these is _olacak_ (and I guess the progressive could be _oluyor_?); in other words, it's behaving in a suppletive way.  And everything about the copula verb is verb-like except its phonology.  Different Turkic languages have or have lost different parts of this copula verb's paradigm, and as you see in Turkish, different pieces are grammaticalised at different levels.  So any criterion to decide which forms should be analysed which ways are somewhat arbitrary (and based on orthography, or at best phonology) and would have to be entirely language-specific.  This seems to incompatible with the goals of UD, but I could be wrong.

(Note: I could see analysing these forms somewhat consistently based on some abstract theory of syntax based on Western European languages, but I think that has very little to do with UD or Turkic languages.)

**Comment by TalhaBedir** [comment]
> Could you explain what you mean by this, and what relevance it has to UD annotation?

Turkish present tense copula is always zero, it never overtly surfaces. You may check: Kornfilt, Jaklin. (1996). On Copular Clitic Forms in Turkish. 

In UD we may only annotate stuff that surfaces. Therefore, if we say _-yim, sin_ etc. in my original post are copulas, that is actually misleading. These are number and person markers _attached_ to a zero copula (that I depicted as [cop]).

The suffixes like -(y)DI, -(y)mIş, -(y)sA, -(y)ken. has copula in them: /y/, although surfacing is phonologically conditioned. Furthermore, these have unattached forms like _idi, imiş, ise, iken_ which has an overt copula /i/.

Therefore we decided on two different paths, although if consistency is key that nuance could be compromised.

**Comment by arademaker** [comment]
Dan said:

> Inconsistencies are pure evil. They prevent comparing data, or applying tools trained on one dataset to another dataset. (And this also extends to other related languages, if it is clear that the phenomenon works exactly the same way there.)

Is it always the case? On the other way around, alternative annotations may help evaluate tools, methods, and linguistic hypotheses. Some annotations maybe are better for some applications and even easier to be learned? 

I do agree that inconsistency inside a treebank is bad. But as UD grows, we will have to face the difficulty of having always agreements between different groups of maintainers. Besides that, as I pointed above, having some alternative approaches can be useful. I would expect that a model trained with one treebank can be better applied to an application given some annotation decisions. Moreover, even from the linguistic research, we could allow alternative approaches to push into different directions of investigations. 

Many interesting discussions in the past are hard to follow because linguistics analysis differs not only in the intrinsic theoretical decisions but also about the formal language used to model those decisions. UD gives us an opportunity, the same metalanguage is being used for many different treebanks. Differences can now be restricted to the theoretical hypothesis being tested.  Putting that in another way, if we ended up with many treebanks equally annotated, what will we learn? Providing treebanks will but just a matter of providing more data without any very real valuable motivation. Not even diversity of linguistics styles is the case since many treebanks already address this.

**Comment by dan-zeman** [comment]
@arademaker : While you can use UD as a meta-language to compare two competing analyses of the same phenomenon in the same language, this should not happen in the treebanks that are included in UD releases. UD is not perfect and this does happen occasionally, but it is not supported and it should be fixed if possible. As a rule, UD has one set of language-specific guidelines per language, and all UD treebanks of that language should follow it. The users can (and often do) assume that two UD-released treebanks of one language are compatible.

**Comment by jonorthwash** [comment]
> In UD we may only annotate stuff that surfaces. Therefore, if we say _-yim, sin_ etc. in my original post are copulas, that is actually misleading. These are number and person markers _attached_ to a zero copula (that I depicted as [cop]).
> 
> The suffixes like -(y)DI, -(y)mIş, -(y)sA, -(y)ken. has copula in them: /y/, although surfacing is phonologically conditioned. Furthermore, these have unattached forms like _idi, imiş, ise, iken_ which has an overt copula /i/.

I might argue that the /j/ in _-(y)Im_ is the same as in _-(y)DI_, etc.—that is, an overt copula marker.  Aside from that, I agree with your understanding of the morphology-to-phonology mapping.  But I don't see how this is relevant to UD.  Do you say there is no copula in _-sIn_ but there is in _-(y)Im_?  And that there's no understanding of 3rd person for the "unmarked" forms, which almost exclusively behave as 3rd person?  Just because there's no overt marking doesn't mean there isn't morphology, or a morphological relationship with other forms in the language.

> Therefore we decided on two different paths, although if consistency is key that nuance could be compromised.

Consistency with what?  Analyses of other Turkic languages?  Annotations of other Turkish treebanks?

**Comment by arademaker** [comment]
> The users can (and often do) assume that two UD-released treebanks of one language are compatible.

Because we say so! But we could educate them differently. Anyway, sorry for the Thread Hijacking

**Comment by TalhaBedir** [comment]
> I might argue that the /j/ in -(y)Im is the same as in -(y)DI, etc.—that is, an overt copula marker. 

Since I follow Kornfilt's paper I referenced above, I do not consider /j/ in -(y)Im is the same as in /j/ in -(y)DI. According to that paper, /j/ in -(y)Im is just a buffer sound as we see in case markers like -(y)A for dative and -(y)I for accusative.

> Do you say there is no copula in -sIn but there is in -(y)Im?

What I say is, there is no overt copula in both of these. And currently, we only mark overt forms.

> Consistency with what? Analyses of other Turkic languages? Annotations of other Turkish treebanks?

Forgive me for the vague explanation, I meant consistency with other Turkish treebanks. Therefore, the split path currently seems the way to go.

**Comment by rueter** [comment]
Sorry for the late question, but in the two Mordvin languages Erzya and Moksha, we use features such as Person[obj] and Person[subj], since we have subject-object conjugation.
On copula nouns, adjectives, spatial adverbs etc. we use Person[subj].
Noting a distinction between deprels of nsubj and nsubj:cop, would you be suggesting the use of Person[cop] in Mordvin languages, too?
